framework/web/UrlRule.php

```diff
@@ -217,6 +217,21 @@ public function parseRequest($manager, $request)
 
         $pathInfo = $request->getPathInfo();
         $suffix = (string) ($this->suffix === null ? $manager->suffix : $this->suffix);
+
+        if (!empty($pathInfo)) {
+            if (substr_compare($pathInfo, '/', -1, 1) === 0) {
+                if ($suffix === '') {
```

Comment:
I am afraid this is not aceeptable: `$suffx` may change from one URL rule to another, but this code will do redirect using suffix from the first rule in the list no matter if it matches URL or not.
This may as well cause the infinite redirection loop if 2 rules in the list use different suffixes.